### PR TITLE
Add 'A brief introduction to BEAM'

### PR DESCRIPTION
**Pull request by jhogberg**
This is a work in progress, posting for early feedback.

### CODE REVIEW

**Comment by KennethL** [comment]
I think it looks good and a very good approach to explain BEAM first. 
In the text the word **emulator** is used in many places and I think we should avoid that word when talking about the BEAM or the Erlang VM. I think emulator is a misleading name for the BEAM or Erlang VM. We should also try to decide if we call it BEAM or Erlang VM.

File: _posts/2020-10-20-a-brief-BEAM-primer.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+sum_tail(List) ->
+    sum_tail(List, 0).
+
+sum_tail([Head | Tail], Acc) ->
+    sum_tail(Tail, Head + Acc);
+sum_tail([], Acc) ->
+    Acc.
+```
+
+Let's look at the instructions one by one:
```
Comment (garazdawi) [inline _posts/2020-10-20-a-brief-BEAM-primer.md]: Maybe add how to get the listing, i.e. by running `erlc -S`

**Comment by jhogberg** [comment]
> I think it looks good and a very good approach to explain BEAM first.
> In the text the word **emulator** is used in many places and I think we should avoid that word when talking about the BEAM or the Erlang VM. I think emulator is a misleading name for the BEAM or Erlang VM. We should also try to decide if we call it BEAM or Erlang VM.

Good point, I wanted to separate BEAM (as in the abstract machine) from ERTS in this post, and used "emulator" for the latter which is probably not the best word for it. I've changed all such uses to "ERTS" instead.

**Comment by HansN** [comment]
> In the text the word emulator is used in many places and I think we should avoid that word when talking about the BEAM or the Erlang VM. I think emulator is a misleading name for the BEAM or Erlang VM. We should also try to decide if we call it BEAM or Erlang VM

I have always thought of BEAM as being one example of an Erlang VM.  The BEAM happens to be the Erlang VM used now in the OTP distribution, but previously we've seen JAM as the Erlang VM in the old versions of OTP.

The *AM are both emulators, while we could have for example source-code level interpreters also as an Erlang VM, like in the very first experiments with Erlang.

So in my opinion the *AM are instances of the class emulators which is a subclass of Erlang VM. Another subclass of Erlang VM is '(source code) interpreter'.

**Pull request merged**